tests/test_example.py

```diff
@@ -1,6 +1,161 @@
+#############################################################################################
+#
+#     You could run test with cmd
+#           python -m pytest -s
+#      or
+#           python -m pytest -s --verbose          to show more info
+#      but don't omit -s in any case!
+#
+#      After starting the tests you will be asked to CHOOSE
+#      dataset together with sessionTimeDay0/1 to pass to the openingsGap function.
+#
+#      Tests 1 to 6 tests Functions ability to detect incorrected input args and raise the relevant error.
+#      These tests use build-in input args and don't depend on the option you choose.
+#
+#      Tests 7 to 12 are testing Function output dataframe itself and in cojunction with input dataframe.
+#      These tests use the input args from the option you choose.
+#
+#############################################################################################
+
+
 def test_example():
     assert True
 
 
-def test_failing_example():
-    assert False
+# def test_failing_example():
+#    assert False
+
+
+import numpy as np
+import pandas as pd
+
+from main import df, empty_dataset, one_day_dataset, openings_gap_inds, openingsGap
+from openingsGap_func import session_time_treatment
+
+
+def test1_interpret_sessionTimeDay():
+    assert session_time_treatment("18m") == np.timedelta64(
+        18, "m"
+    ), "sessionTimeDay arg wrong interpretation"
+    assert session_time_treatment("3h") == np.timedelta64(
+        3, "h"
+    ), "sessionTimeDay arg wrong interpretation"
+    assert session_time_treatment("1D") == np.timedelta64(
+        1, "D"
+    ), "sessionTimeDay arg wrong interpretation"
+
+
+def test2_empty_init_df():
+    try:
+        openingsGap(empty_dataset)
+    except Exception as e:
```

Comment:
I thought the assert comment is enough, but okey.
Done for test functions names.

Related to  your suggestion to change the function body to
with pytest.raises(ValueError) as e_info:
    ...
I check two cases:
- the core function doesn't raise an error at all
- the core function raises the error, but with wrong text comment.
And I don't understand how to realize this with suggested construction.

Reply:
Ah I see. The case where the core function doesn't raise an error is captured by pytest specifically. It will fail the test no ValueError is thrown. And the assert should check that the error text is correct. 

[reference](https://stackoverflow.com/questions/23337471/how-to-properly-assert-that-an-exception-gets-raised-in-pytest)

Reply:
This test doesn't work. I tried to rewrite test according to the suggestion in two versions. Than I removed the empty df check in the openingGap func. But in this case pytest doesn't work because before it openingGap goes forward and raise another error (it is not able to treat the empty df).
![image](https://github.com/AITRdevelopment/openingGap/assets/115402114/4ef58c2e-ccf3-470d-a5ca-110313f1ddbf)
![image](https://github.com/AITRdevelopment/openingGap/assets/115402114/0194fb82-9b0f-4920-a3b4-cd8bc871fee1)
![pic](https://github.com/AITRdevelopment/openingGap/assets/115402114/75456f11-5948-41e3-8b91-fb8c4ae56b08)

Reply:
It seems I was not right, the pytest shows the failure, but it is followed by error raised  when openingGap func continues running. If such behavior is ok for you, please let me know and I will modify all tests in this way.